packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -53,7 +56,7 @@ class _InsightsHandler {
     private _currentTestId: string | undefined
     private _cbtQueue: Array<CBTData> = []
 
-    constructor (private _browser: WebdriverIO.Browser | WebdriverIO.MultiRemoteBrowser, isAppAutomate?: boolean, private _framework?: string, _userCaps?: Capabilities.ResolvedTestrunnerCapabilities) {
+    constructor (private _browser: WebdriverIO.Browser | WebdriverIO.MultiRemoteBrowser, isAppAutomate?: boolean, private _framework?: string, _userCaps?: Capabilities.ResolvedTestrunnerCapabilities, _options?: BrowserstackConfig & BrowserstackOptions) {
```

Comment:
Can we refactor `isAppAutomate` as well as the function doesn't require any scope and only takes a capability object. There is not reason to pass this into the class if you have the capabilities given here as well

Reply:
I checked the capabilities object we have inside this class & that does not have any key value pair we can use to identify if the session is an app automation session. Hence we're making use of the `isAppAutomate` boolean.

Reply:
Actually since the browser & the capability object are both passed here, we can simply move that method over here instead of passing the boolean inside the constructor.